extensions/blocks/social-previews/modal.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Social Previews modal component.
+ *
+ * Shows individual previews in modal window.
+ */
+
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Modal, TabPanel } from '@wordpress/components';
+import { Fragment } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import SocialPreviewsUpgrade from './upgrade';
+import { AVAILABLE_SERVICES } from './constants';
+import { SocialServiceIcon } from '../../shared/icons';
+import { getMediaSourceUrl } from './utils';
+import './modal.scss';
+
+const SocialPreviewsModal = function SocialPreviewsModal( {
+	onClose,
+	showUpgradeNudge,
+	image,
+	title,
+	description,
+	url,
+	author,
+} ) {
+	// Inject the service icon into the title
+	const tabs = AVAILABLE_SERVICES.map( service => {
+		return {
+			...service,
+			title: (
+				<Fragment>
+					<SocialServiceIcon serviceName={ service.icon } />
+					{ service.title }
+				</Fragment>
+			),
+		};
+	} );
+
+	return (
+		<Modal
+			onRequestClose={ onClose }
+			title={ __( 'Social Previews', 'jetpack' ) }
+			className="jetpack-social-previews__modal"
+		>
+			{ showUpgradeNudge ? (
+				<SocialPreviewsUpgrade />
+			) : (
+				<TabPanel
+					className="jetpack-social-previews__modal-previews"
+					tabs={ tabs }
+					orientation="vertical"
+				>
+					{ tab => (
+						<div>
+							<tab.preview
+								title={ title }
+								description={ description }
+								url={ url }
+								author={ author }
+								image={ image }
+							/>
+						</div>
+					) }
+				</TabPanel>
+			) }
+		</Modal>
+	);
+};
+
+export default withSelect( ( select, props ) => {
+	// No need to load anything when the feature is not active.
+	if ( props.showUpgradeNudge ) {
+		return {};
+	}
+
+	const { getMedia, getUser } = select( 'core' );
+	const { getCurrentPost, getCurrentPostId, getEditedPostAttribute } = select( 'core/editor' );
+
+	const featuredImageId = getEditedPostAttribute( 'featured_media' );
+	const authorId = getEditedPostAttribute( 'author' );
+	const user = authorId && getUser( authorId );
+
+	return {
+		post: getCurrentPost(),
+		title: getEditedPostAttribute( 'title' ),
+		description: getEditedPostAttribute( 'excerpt' ) || getEditedPostAttribute( 'content' ),
+		url: getEditedPostAttribute( 'link' ),
+		author: user?.name,
+		image:
+			!! featuredImageId && getMediaSourceUrl( getMedia( featuredImageId ), getCurrentPostId() ),
```

Comment:
Not for this PR, but we'll probably need to adjust this to be more exact; Social Previews are generated from Open Graph Meta Tags and Twitter Meta Tags, and those are not always as simple as checking for a featured image or an excerpt. That depends on what plugin you're using to generate those tags. 

In Jetpack and on WordPress.com for example, we go through different steps to find an image, and do not limit ourselves to featured images.